### DISCUSSION TITLE
Simulation of NaCl dissolution with reactiveTransportDBSFoam

### DISCUSSION THREAD

**Discussion by Mahsa-Hs** [GeoChemFoam/GeoChemFoam#20] [Q&A] [answered]
I want to simulate the dissolution rate of a spherical NaCl particle in water using reactiveTransportDBSFoam. The reaction will be NaCl--> Na+ + Cl- with R=k(1-[Na+][Cl-]/Keq).  I changed the thermoPhysicalProperties as below:

solutionSpecies
{
   Na+
   {
      D    D  [0 2 -1 0 0 0 0] 1.33e-9;
   }
   Cl-
   {
      D    D [0 2 -1 0 0 0 0] 2.03e-9;
   }
};
kineticPhaseReactions
{
}
kineticPhases
{
  Halite
  {
      species
      {
          Na+;
          Cl-;
      }
       Ke Ke [0 -2 -1 0 1 0 0] 0.25;
  }
};

Also, I defined Na+ and Cl- in the 0 directory. I want to simulate this case without any flow. The dissolution will be done only by diffusion process.  I set two different initial concentration (0 and 0.01) for Na+ and Cl- in internalField in two different cases. Other boundaries had zeroGradient condition. However, the simulations are not successful. There was not any error, but when viewing results in paraview there are not any changes in the concentration of Na+, Cl- and eps field. What should I set for Na+ and Cl- in internalField values and what other changes should I make? 
Thanks for your help.

**Comment by hannahmenke** [comment, marked as answer]
First turn flow back on to make sure it is working with flow. Then check if the timescale of your run matches the expected timescale of your diffusion.

**Comment by julienmaes** [comment]
There is no kineticPhaseReactions in your thermoPhysicalProperties, this is were you define the reaction , stoichiometric coeff and reaction rate. Convert the one from the test case (calcite) into Halite

**Reply by Mahsa-Hs** [reply]
I just included kineticPhases as the reaction was equilibrium, and wrote Ke. Should I have both?

**Reply by julienmaes** [reply]
reactiveTransportDBSFoam does not solve dissolution/precipitation reactions as equilibrium because they are not usually at equilibrium at the pore-scale. So yes, you need to include a reaction constant and the reaction will be Na++Cl---> NaCl with reaction rate k*(1-Omega)
